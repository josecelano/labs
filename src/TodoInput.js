/**
 * This component should render a controlled input for text
 */
import React from 'react';

const TodoInput = (props) => (
    <form onSubmit={props.handleNewItem}>
        <label>
            <input type="text" value={props.value} onChange={props.onChange}/>
        </label>
    </form>
)

export default TodoInput;
