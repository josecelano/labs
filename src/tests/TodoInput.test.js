/**
 * Tests for the TodoInput
 */
import React from 'react';
import renderer from 'react-test-renderer';
import TodoInput from '../TodoInput';

describe('<TodoInput />', () => {
    it('should render an input box', () => {
        const tree = renderer
            .create(<TodoInput />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
