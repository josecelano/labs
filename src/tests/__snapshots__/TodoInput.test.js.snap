exports[`test should render an input 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="text"
      value="hello" />
  </label>
</form>
`;
